dds/DCPS/DCPS_Utils.cpp

```diff
@@ -464,21 +477,35 @@ bool repr_to_encoding_kind(DDS::DataRepresentationId_t repr, Encoding::Kind& kin
   case DDS::XCDR2_DATA_REPRESENTATION:
     kind = Encoding::KIND_XCDR2;
     break;
+  case OpenDDS::DCPS::UNALIGNED_CDR_DATA_REPRESENTATION:
+    kind = Encoding::KIND_UNALIGNED_CDR;
+    break;
   default:
     return false;
   }
   return true;
 }
 
-DDS::DataRepresentationIdSeq get_effective_data_rep_qos(const DDS::DataRepresentationIdSeq& qos, bool reader)
+DDS::DataRepresentationIdSeq get_writer_effective_data_rep_qos(const DDS::DataRepresentationIdSeq& qos,
+  bool cdr_encapsulated)
 {
   if (qos.length() == 0) {
     DDS::DataRepresentationIdSeq ids;
-    ids.length(reader ? 2 : 1);
-    ids[0] = DDS::XCDR2_DATA_REPRESENTATION;
-    if (reader) {
-      ids[1] = DDS::XCDR_DATA_REPRESENTATION;
-    }
+    ids.length(1);
+    ids[0] = cdr_encapsulated ? DDS::XCDR2_DATA_REPRESENTATION : OpenDDS::DCPS::UNALIGNED_CDR_DATA_REPRESENTATION;
+    return ids;
+  }
+  return qos;
+}
+
+DDS::DataRepresentationIdSeq get_reader_effective_data_rep_qos(const DDS::DataRepresentationIdSeq& qos)
+{
+  if (qos.length() == 0) {
+    DDS::DataRepresentationIdSeq ids;
+    ids.length(3);
+    ids[0] = DDS::XCDR_DATA_REPRESENTATION;
+    ids[1] = DDS::XCDR2_DATA_REPRESENTATION;
```

Comment:
Prior to this change, XCDR2 came first in the sequence.  I'm not sure it matters but it seems like we should preserve that unless there's a reason not to.